### ISSUE TITLE
Objects mutation and vue reactivity

### ISSUE THREAD

**Issue by Waserman** [vuejs/vuex#827]
### Version
2.3.0

### Reproduction link
[https://github.com/Waserman/vue-reactivity-experiment](https://github.com/Waserman/vue-reactivity-experiment)

### Steps to reproduce
The README file is very straightforward.
Follow the instructions over there.

### What is expected?
I expect store mutations to be rendered in the view

### What is actually happening?
store mutation is not reactive when mutating objects and objects inside arrays.

---
I used all the approaches listed in your documentation (you could see those attempts commented out in the code).

**Comment by LinusBorg** [comment]
Everything seems to work fine for me ...

Whenever I click "update" for any of the three, the price of all items changes to something random, as it should 

Yes, only for the "associative array" that new "update" property is shown, but that's simply because you only render it for that collection in the template, so I don't see what other behaviour might be expected:
```html
<li class="list-group-item" v-if="stock.updated">update: {{stock.updated}}</li>
```

Am I missing something?

**Comment by Waserman** [comment]
OMG I feel so embarrassed. Apparantly I missed somthing (!!)
In my application I use custom component for the rendering.
While I made this sample application, for the sake simplicity I did not used component, and must have forgotten to render the new 'updated' property for the other two collections ^^'

during composing this issue and coding the sample app, I made some research and used some approaches I didn't use in my application (the original one). It is likely that one of those solved the problem...

If somthing good went out of this unnecessary issuing is the comprehensive example of how one should do collections mutations in vue. I came across tons of questions over stackoverflow about this topic. Now I think I have great reference :)

thanks a lot!
